### PR TITLE
In TBranchElement init offset avoid stripping part of the name.

### PR DESCRIPTION
**Pull request by pcanal**
A prefix needs to be followed by a delimiter

This fixes ROOOT-5983.

### CODE REVIEW

**Comment by eguiraud** [comment]
@phsft-bot build please

File: tree/tree/src/TBranchElement.cxx [hunk 1]
```diff
@@ -53,12 +53,14 @@ ClassImp(TBranchElement);
...
+   void RemovePrefix(TString& str, const TString &prefix) {
       // -- Remove a prefix from a string.
-      if (str.Length() && prefix && strlen(prefix)) {
-         if (!str.Index(prefix)) {
+      // -- Require a '.' after the prefix.
+      if (prefix.Length() && prefix.Length() <= str.Length()
+          && (str.Data()[prefix.Length()] == '.' || (prefix[prefix.Length()-1]=='.')))
+      {
+         if (!str.Index(prefix))
             str.Remove(0, strlen(prefix));
```
Comment (eguiraud) [inline tree/tree/src/TBranchElement.cxx:63]:
```suggestion
            str.Remove(0, prefix.Length());
```

?

**Review by eguiraud — APPROVED** [review]
LGTM!

**Comment by eguiraud** [comment]
@pcanal can merge?

**Comment by pcanal** [comment]
Errors are unrelated.

**Pull request merged**